### PR TITLE
Modal component - Added `backdrop='static'` option to allow clicking outside the Modal content

### PR DESCRIPTION
**Pull request by pankaj-mahendran**
## Description

- In the modus modal component a new property "backdrop" is required.
- This would accept 3 values - true, false, "static"
- When the backdrop is "static" - the modal does not close when we click outside the modal.
- When the backdrop is false - the background of the modal is not an opaque screen, the background would be the same page, The modal would not also be closed once clicked outside the modal.
- When the backdrop is true ( default mode ) - the backdrop is an opaque image, as usual and the modal would close once clicked outside the modal.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## How Has This Been Tested?

- Tried integrating the modus modal without backdrop property, works fine.
- Integrated with backdrop property **static**, the modal did not close when clicked anywhere outside the modal.
-  Integrated with backdrop property **false**, the modal's backdrop is the component behind the modal and the modal did not close when clicked anywhere outside the modal.
-  Integrated with backdrop property **true** ( existing/default behavior), the modal's backdrop is an opaque screen and the modal did close when clicked anywhere outside the modal.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by msankaran0712** [comment]
As discussed @pankaj-mahendran we need to remove backdrop="false" option and if possible add unit tests

**Comment by pankaj-mahendran** [comment]
Removed backdrop=false and added unit tests in my latest commit.

File: stencil-workspace/src/components/modus-modal/modus-modal.tsx [hunk 1]
```diff
@@ -68,12 +71,15 @@ export class ModusModal {
   }
 
   handleOverlayClick(event: MouseEvent): void {
-    if (this.ignoreOverlayClick || !(event.target as HTMLElement).classList.contains('overlay')) {
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-modal/modus-modal.tsx:71]: we need these three lines

**Review by msankaran0712 — APPROVED** [review]

**Review by msankaran0712 — CHANGES_REQUESTED** [review]
Please rebase the branch from `main`

**Review by msankaran0712 — APPROVED** [review]

**Pull request merged**